ledger/test/test_compact_serializer.py

```diff
@@ -0,0 +1,228 @@
+from collections import OrderedDict
+
+from ledger.serializers.compact_serializer import CompactSerializer
+
+fields = OrderedDict([
+    ("f1", (str, str)),
+    ("f2", (str, str)),
+    ("f3", (str, int)),
+    ("f4", (str, float)),
+    ("f5", (str, eval))
+])
+
+def testInitCompactSerializerWithCorrectFileds():
+    CompactSerializer(fields)
+
+def testInitCompactSerializerNoFileds():
+    CompactSerializer()
+
+def testInitCompactSerializerEmptyFileds():
+    fields = OrderedDict([])
+    CompactSerializer(fields)
+
+def testSerializeSimpleJson():
+    serializer = CompactSerializer(fields)
+    assert b"v1|v2|3|4.0|True" == \
+           serializer.serialize(
+               {"f1": "v1", "f2": "v2", "f3": 3, "f4": 4.0, "f5": True}
+           )
+
+
+def testDeserializeSimpleJson():
+    serializer = CompactSerializer(fields)
+
+    assert {"f1": "v1", "f2": "v2", "f3": 3, "f4": 4.0, "f5": True} == \
+            serializer.deserialize(b"v1|v2|3|4.0|True")
+
+
+def testSerializeDeserializeSimpleJson():
+    serializer = CompactSerializer(fields)
+
+    json = {"f1": "v1", "f2": "v2", "f3": 3, "f4": 4.0, "f5": True}
+    serialized = serializer.serialize(json)
+    deserialized = serializer.deserialize(serialized)
+
+    assert json == deserialized
+
+
+def testSerializeToBytes():
+    serializer = CompactSerializer(fields)
+    assert b"v1|v2|3|4.0|True" == \
+           serializer.serialize(
+               {"f1": "v1", "f2": "v2", "f3": 3, "f4": 4.0, "f5": True},
+               toBytes=True
+           )
+
+def testSerializeToString():
+    serializer = CompactSerializer(fields)
+    assert "v1|v2|3|4.0|True" == \
+           serializer.serialize(
+               {"f1": "v1", "f2": "v2", "f3": 3, "f4": 4.0, "f5": True},
+               toBytes=False
+           )
+
+def testDeserializeFromBytes():
```

Comment:
A good practice of testing is to test small pieces of code (serialization and deserialization in this case)